Add samples for reading and writing packages using the API (#2126)
NuGet/docs.microsoft.com-nuget: docs/reference/NuGet-Client-SDK.md

## Getting started

### Install the packages

```ps1

> dotnet add package NuGet.Protocol  # interact with HTTP and folder based NuGet package feeds, includes NuGet.Packaging

Reviewer: ```suggestion
dotnet add package NuGet.Protocol  # interact with HTTP and folder-based NuGet package feeds, includes NuGet.Packaging
```

[!code-csharp[SearchPackages](~/../nuget-samples/NuGetProtocolSamples/Program.cs?name=SearchPackages)]

### Create a package

Create a package, set metadata, and add dependencies using [`NuGet.Packaging`](https://www.nuget.org/packages/NuGet.Packaging).

> [!IMPORTANT]
> It is strongly recommended that NuGet packages are created using the official NuGet tooling and **not** using this

> > low level API. There are a variety of characteristics important for a well-formed package and the latest version of

Reviewer: ```suggestion
> low-level API. There are a variety of characteristics important for a well-formed package and the latest version of
```

> tooling helps incorporate these best practices.
> 
> For more information about creating NuGet packages, see the overview of the
> [package creation workflow](../create-packages/overview-and-workflow.md) and the documentation for official pack

> > tooling (e.g. [using the dotnet CLI](../create-packages/creating-a-package-dotnet-cli.md)).

Reviewer: ```suggestion
> tooling (For example, [Using the dotnet CLI](../create-packages/creating-a-package-dotnet-cli.md)).
```

> > tooling (For example, [Using the dotnet CLI](../create-packages/creating-a-package-dotnet-cli.md)).

Reviewer 2: The upper casing seems odd to me. What do you think of this?

```suggestion
> tooling (for example, [using the dotnet CLI](../create-packages/creating-a-package-dotnet-cli.md)).
```

I feel less strongly about `Using the dotnet CLI`, feel free to keep that as-is.

Reviewer: @loic-sharma You're right about the "for example". The reference to another topic ("Using the...") is capitalized, just like the topic.

> [!code-csharp[CreatePackage](~/../nuget-samples/NuGetProtocolSamples/Program.cs?name=CreatePackage)]

Reviewer: The sample isn't rendering. Path issue?

Author: This PR needs to be completed first:
https://github.com/NuGet/Samples/pull/40

Author: This should be fixed now.

### Read a package

Read a package from a file stream using [`NuGet.Packaging`](https://www.nuget.org/packages/NuGet.Packaging).

> [!code-csharp[ReadPackage](~/../nuget-samples/NuGetProtocolSamples/Program.cs?name=ReadPackage)]

Reviewer: Sample not rendering

Author: This PR needs to be completed first:
https://github.com/NuGet/Samples/pull/40
